src/gocam/cli.py

```diff
@@ -476,5 +480,273 @@ def flatten_models(input_file, input_format, output_format, output_file, fields)
+                model = MinervaWrapper.minerva_object_to_model(minerva_dict)
+                return model
+        except Exception as e:
+            logger.error(f"Failed to load model from {file_path}: {e}")
+            return None
+
+    def translate_to_networkx(model: Model, output_path: str, model_filename: str):
+        """Translate a model to NetworkX format and save as JSON."""
+        try:
+            # Initialize the translator
+            indexer = Indexer()
+            translator = ModelNetworkTranslator(indexer=indexer)
+            
+            # Translate the model
+            json_output = translator.translate_models_to_json([model], include_model_info=True, indent=2)
+            
+            # Save to file
+            output_file = os.path.join(output_path, f"{model_filename}_networkx.json")
+            with open(output_file, 'w') as f:
+                f.write(json_output)
+            
+            logger.debug(f"Saved NetworkX translation to {output_file}")
+            
+        except Exception as e:
+            logger.error(f"Failed to translate {model_filename} to NetworkX: {e}")
+
+    def translate_to_cx2(model: Model, output_path: str, model_filename: str):
+        """Translate a model to CX2 format and save as JSON."""
+        try:
+            # Translate the model
+            cx2_data = model_to_cx2(model)
+            
+            # Save to file
+            output_file = os.path.join(output_path, f"{model_filename}_cx2.json")
+            with open(output_file, 'w') as f:
+                json.dump(cx2_data, f, indent=2)
+            
+            logger.debug(f"Saved CX2 translation to {output_file}")
+            
+        except Exception as e:
+            logger.error(f"Failed to translate {model_filename} to CX2: {e}")
+
+    # Ensure formats is a list (Click's multiple option sometimes returns a tuple)
+    if isinstance(formats, tuple):
+        formats = list(formats)
+    
+    # Create output directories
+    output_paths = {}
+    if "networkx" in formats:
+        output_paths["networkx"] = output_networkx
+        os.makedirs(output_networkx, exist_ok=True)
+        click.echo(f"Created output directory for NetworkX: {output_networkx}", err=True)
+    
+    if "cx2" in formats:
+        output_paths["cx2"] = output_cx2
+        os.makedirs(output_cx2, exist_ok=True)
+        click.echo(f"Created output directory for CX2: {output_cx2}", err=True)
+    
+    # Create temporary directory for extraction
+    with tempfile.TemporaryDirectory() as temp_dir:
+        try:
+            # Download and extract the tarball
+            extracted_dir = download_and_extract_tarball(url, temp_dir)
+            
+            # Find all JSON model files
+            json_files = find_json_models(extracted_dir)
+            
+            if limit:
+                json_files = json_files[:limit]
+                click.echo(f"Limited processing to {len(json_files)} models", err=True)
```

Comment:
If performance is a concern, I would pass `limit` to the `find_json_models` function so that it can stop iterating after `limit` files have been found.

If performance is _really_ a concern, I would `yield` paths from `find_json_models` instead of returning a list and use `islice` (https://docs.python.org/3/library/itertools.html#itertools.islice) to apply the `limit`.